tests/test_gen_gitreview.py

```diff
@@ -0,0 +1,75 @@
+from unittest import mock, TestCase
+
+import toolchest.gen_gitreview
+
+
+class test_generate_gitreview(TestCase):
+
+    def test_gen_single_gitreview(self):
+        """
+        GIVEN a list of .gitreview fields including a defaultrebase value
+        WHEN generate_gitreview is called
+        THEN a .gitreview file is written with the specified fields
+        AND the given defaultrebase value is written
+        """
+        mocked = mock.mock_open()
+        path = "/tmp/testing_tmp"
+        host = "random_host_str"
+        host_port = "1234"
+        project = "test_project"
+        branch = "test_branch_name"
+        remote = "test_remote"
+        rebase = 0
+
+        with mock.patch("toolchest.gen_gitreview.open", mocked, create=True):
```

Comment:
Why not just write an actual file to the pytest tmpdir and then validate the contents?

Reply:
I thought that maybe it would be best to mock the calls even though it would be very straightforward to actually write to a file. Is it preferred to do it one way or another?

Reply:
There may be a reason I am not thinking of to not do this (@jpichon can comment if she has one), but I cnnot think of what it would be.  Pytest provided a temporary directory [fixture](https://docs.pytest.org/en/7.1.x/getting-started.html#request-a-unique-temporary-directory-for-functional-tests) for just this purpose. I think it would be much simpler and clearer to test the real functionality, and save mocks for cases where we have an external service that we don't want to actually invoke. This is the guideline I use myself - if you can test the real thing do so, only bring in mocks when it would cause either an external dependency, or return inconsistent results (and very often those are related).

Reply:
Agreed that using a fixture to write an actual temporary file rather than mocking the calls should look cleaner and clearer.